python/lsst/sims/maf/metricBundles/metricBundleGroup.py

```diff
@@ -465,48 +465,49 @@ def readAll(self):
         reduceBundleDict = {}
         for b in self.bundleDict.itervalues():
             filename = os.path.join(self.outDir, b.fileRoot+'.npz')
-            # Create a temporary metricBundle to read the data into.
-            #  (we don't use b directly, as this overrides plotDict/etc).
-            tmpBundle = createEmptyMetricBundle()
-            tmpBundle.read(filename)
-            # Copy the tmpBundle metricValues into b.
-            b.metricValues = ma.copy(tmpBundle.metricValues)
-            del tmpBundle
-
-            # Look to see if this is a complex metric, with associated 'reduce' functions, and read those in.
-            if len(b.metric.reduceFuncs) > 0:
-                origMetricName = b.metric.name
-                for reduceFunc in b.metric.reduceFuncs.itervalues():
-                    reduceName = origMetricName + '_' + reduceFunc.__name__.replace('reduce', '')
-                    # Borrow the fileRoot in b (we'll reset it appropriately afterwards).
-                    b.metric.name = reduceName
+            if os.path.isfile(filename):
```

Comment:
I'm not sure what happened to github here - the 'diff' seems like it should just be one of whitespace, but github has also moved a bunch of lines around. Very confusing. 

On an actual question: 
the metricBundle itself should check to see if the file exists on 'read' and it seems like what we should actually do here is try/except on the read. And handle the non-existence of files gracefully (except would be print a warning about that particular metricBundle not existing on disk and continuing, I guess). This warning doesn't happen now - you just get a silent failure. 

Also - when we do this for the reduce files, we should do the same thing (try/except and print an error message if it doesn't exist). 

This has the slight added advantage of not adding extra whitespace which we don't really need (which we've seen from various pieces of code can be problematic).